pkg/cloudprovider/cloudprovider.go

```diff
@@ -165,6 +169,50 @@ func (c *CloudProvider) Delete(ctx context.Context, node *v1.Node) error {
 	return c.instanceProvider.Terminate(ctx, node)
 }
 
+func (c *CloudProvider) IsMachineDrifted(ctx context.Context, machine *corev1alpha1.Machine) (bool, error) {
+	// Not needed when GetInstanceTypes removes provisioner dependency
+	provisioner := &v1alpha5.Provisioner{}
+	if err := c.kubeClient.Get(ctx, types.NamespacedName{Name: machine.Labels[v1alpha5.ProvisionerNameLabelKey]}, provisioner); err != nil {
+		return false, fmt.Errorf("getting provisioner, %w", err)
+	}
+	instanceTypes, err := c.GetInstanceTypes(ctx, provisioner)
+	if err != nil {
+		return false, fmt.Errorf("getting instanceTypes, %w", err)
+	}
+	nodeInstanceType := lo.Filter(instanceTypes, func(instType *cloudprovider.InstanceType, _ int) bool {
+		return instType.Name == machine.Labels[v1.LabelInstanceTypeStable]
+	})
+	if len(nodeInstanceType) == 0 {
+		return false, fmt.Errorf("getting nodeInstanceType")
+	}
+	nodeTemplate, err := c.resolveNodeTemplate(ctx, []byte(machine.
+		Annotations[v1alpha5.ProviderCompatabilityAnnotationKey]), machine.
+		Spec.MachineTemplateRef)
+	if err != nil {
+		return false, fmt.Errorf("resolving node template, %w", err)
```

Comment:
What if we just returned false if no node template?

Reply:
Returning false if IsNotFound.

Reply:
```suggestion
		return false, k8sClient.IgnoreNotFound(err)
```